lib/scholar/naive_bayes/gaussian_naive_bayes.ex

```diff
@@ -0,0 +1,442 @@
+defmodule Scholar.NaiveBayes.Gaussian do
```

Comment:
Given that Scholar also houses non-classifier stuff like interpolation algorithms, how about using `Scholar.Classifiers.NaiveBayes` as a namespace?

There might be other things that could move into this `Classifiers` namespace, though, in a separate PR

Reply:
I believe our goal is to mirror Scipy as much as possible, no? Or do you mean we could have both `Scholar.Classifiers.NaiveBayes.Gaussian` and `Scholar.SomethingElse.NaiveBayes.Gaussian`?

Reply:
I didn't find Naive Bayes in SciPy, but there is sklearn.naive_bayes. There, they also suffix the submodules like sklearn.naive_bayes.GaussianNB.

Not sure if I like it, but it's your call

Reply:
@polvalente it is up to us (you?) to decide because honestly I am not fully qualified. Can you look at the sklearn.naive_bayes namespace and say if those are all classifiers or if each of them belong to a different category and therefore effectively deserve their own namespace?

Reply:
All of the algorithms in sklearn.naive_bayes are classification algorithms based on the naïve Bayes concept/methodology (i.e. assumption that features are all independent random variables).

I'd put them in Scholar.Classification.NaiveBayes and also move Scholar.Linear.LogisticRegression to Scholar.Classification.Linear.LogisticRegression.

LogisticRegression by itself isn't necessarily a classification algorithm, but as it is implemented here, it is a classifier based on logistic regression. Same for NaiveBayes, they are classifiers based on the naïve interpretation of Bayes' theorem.

Reply:
Hrm... Honestly, I would like to avoid four levels of nesting. It is not very common in Elixir, especially in public APIs. 

Also, is it possible that we would augment LogisticRegression APIs in the future so it does more than classification?